pennylane/qchem/factorization.py

```diff
@@ -274,44 +274,151 @@ def basis_rotation(one_electron, two_electron, tol_factor=1.0e-5):
     """
 
-    t_matrix = one_electron - 0.5 * np.einsum("illj", two_electron)
-    two_electron = np.swapaxes(two_electron, 1, 3)
+    num_orbitals = one_electron.shape[0] * 2
+    one_body_tensor = np.kron(one_electron, np.eye(2))  # account for spin
+    one_body_correction, chemist_two_body_tensor = chemist_transform(two_electron, True)
 
-    _, eigvals_m, eigvecs_m = qml.qchem.factorize(two_electron, tol_factor, 0.0)
-    t_eigvals, t_eigvecs = np.linalg.eigh(t_matrix)
+    chemist_one_body_tensor = one_body_tensor + one_body_correction
+    t_eigvals, t_eigvecs = np.linalg.eigh(chemist_one_body_tensor)
 
-    eigvals = [np.array(t_eigvals)] + [np.outer(x, x).flatten() * 0.5 for x in eigvals_m]
-    eigvecs = [t_eigvecs] + eigvecs_m
+    factors, _, _ = factorize(chemist_two_body_tensor, tol_factor=tol_factor)
+    factors = [np.kron(factor, np.eye(2)) for factor in factors]  # account for spin
+
+    v_coeffs, v_unitaries = np.linalg.eig(factors)
+    indices = [np.argsort(v_coeff)[::-1] for v_coeff in v_coeffs]
+    v_coeffs = [v_coeff[indices[idx]] for idx, v_coeff in enumerate(v_coeffs)]
+    v_unitaries = [v_unitary[:, indices[idx]] for idx, v_unitary in enumerate(v_unitaries)]
 
     ops_t = 0.0
-    for i in range(len(eigvals[0])):
-        ops_t += 0.5 * eigvals[0][i] * qml.Identity(i) - 0.5 * eigvals[0][i] * qml.PauliZ(i)
+    for p in range(num_orbitals):
+        ops_t += 0.5 * t_eigvals[p] * (qml.Identity(p) - qml.PauliZ(p))
 
     ops_l = []
-    for coeff in eigvals[1:]:
-        n = int(len(coeff) ** 0.5)
+    for idx in range(len(factors)):
         ops_l_ = 0.0
-        count = 0
-        for i in range(n):
-            for j in range(n):
-                c = coeff[count]
-                count += 1
+        for p in range(num_orbitals):
+            for q in range(num_orbitals):
                 ops_l_ += (
-                    c
+                    v_coeffs[idx][p]
+                    * v_coeffs[idx][q]
                     * 0.25
                     * (
-                        qml.Identity(i)
-                        - qml.PauliZ(i)
-                        - qml.PauliZ(j)
-                        + qml.pauli.pauli_mult_with_phase(qml.PauliZ(i), qml.PauliZ(j))[0]
+                        qml.Identity(p)
+                        - qml.PauliZ(p)
+                        - qml.PauliZ(q)
+                        + qml.pauli.pauli_mult_with_phase(qml.PauliZ(p), qml.PauliZ(q))[0]
                     )
                 )
         ops_l.append(ops_l_)
 
     ops = [ops_t] + ops_l
-
-    c_group = [op.coeffs * 2 for op in ops]  # coeffs are multiplied by 2 to account for spin
+    c_group = [op.coeffs for op in ops]
     o_group = [op.ops for op in ops]
-    u_transform = [eigvec.T for eigvec in eigvecs]
+    u_transform = list([t_eigvecs] + list(v_unitaries))  # Inverse of diagonalizing unitaries
 
     return c_group, o_group, u_transform
+
+
+def chemist_transform(two_body_tensor, spatial_basis=True):
```

Comment:
Is the one-body correction needed anywhere just by itself? This function could take the one- and two-body terms, in physicist notation, and return them in the chemist notation. So, it will take `h_pq` and `h_pqrst` and returns `T_pq` and `V_pqrst`.

Reply:
I am not sure, but I have updated the function to return both to provide users with flexibility.